internal/config/config.go

```diff
@@ -63,15 +63,18 @@ var AWS = &config.AWS
 var Features = &config.FeatureFlags
 
 func init() {
-	var err error
-
 	viper.SetDefault("logging.level", 1)
 	viper.SetDefault("database.host", "localhost")
 	viper.SetDefault("database.port", 5432)
 	viper.SetDefault("database.loglevel", 1)
 	viper.SetDefault("cloudwatch.enabled", false)
 	viper.SetDefault("prometheus.port", 9000)
 
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
```

Comment:
One thing that looks nasty is ordering of init() calls, I am not able to tell if this line is called after or before `ENV_config.go`. This could lead to some nasty bugs. It looks like that Go compiler calls them in lexicographic order: https://go.dev/ref/spec#Package_initialization

So you were simply lucky this time but I suggest to extract all the three init functions into separate files and naming it the following way:

* 0_config_common.go
* 5_config_test.go
* 5_config_development.go

This way it is crystal clear even after 5 years what is going on here.

Reply:
I've added numbered prefixes, but I decided to prefix with environment, because config_test clashes with test file for config. :(